### PR TITLE
[mac] enable broadcast message retransmission in MAC layer

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: src/core/openthread-core-default-config.h [hunk 1]
```diff
@@ -159,6 +159,16 @@
 #define OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_POLLS 4
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAX_TRANSMIT_NUM_BROADCAST
+ *
+ * Maximum number of MAC layer transmission for a broadcast frame.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MAX_TRANSMIT_NUM_BROADCAST
```
Comment (jwhui) [inline src/core/openthread-core-default-config.h]: This specifies how many times the MAC layer will transmit every time, not some number between 0 and MAX.  How about calling this something like `OPENTHREAD_CONFIG_TX_NUM_BCAST`.

**Comment by jwhui** [comment]
@zhanglongxia, thanks for this update.  Handling duplicate MLE messages at the MLE layer is a much better approach than attempting to do the same at the MAC layer.

File: src/core/mac/mac.hpp [hunk 2]
```diff
@@ -95,6 +95,13 @@ enum
      *
      */
     kIndirectFrameMacTxAttempts = OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_PER_POLL,
+
+    /**
+     *
```
[the remark this answers is not in the document]
Comment (zhanglongxia) [reply src/core/mac/mac.hpp]: Done

File: src/core/openthread-core-default-config.h (hunk 1 above)
Comment (zhanglongxia) [reply src/core/openthread-core-default-config.h]: Done

**Comment by zhanglongxia** [comment]
@jwhui, thanks for abtink's suggestion. Current method is much easier and better. I have updated the code for your comments. Please review it again.

File: src/core/thread/mle.cpp [hunk 3]
```diff
@@ -1795,13 +1795,14 @@ void Mle::RemoveDelayedDataResponseMessage(void)
     {
         delayedResponse.ReadFrom(*message);
 
-        if (message->GetSubType() == Message::kSubTypeMleDataResponse)
+        if (message->GetSubType() == aSubType &&
+            (aDestination.IsMulticast() || delayedResponse.GetDestination() == aDestination))
```
[the remark this answers is not in the document]
Comment (zhanglongxia) [reply src/core/thread/mle.cpp]: Updated

File: src/core/thread/mle_router.cpp [hunk 4]
```diff
@@ -1580,6 +1580,13 @@ otError MleRouter::HandleParentRequest(const Message &aMessage, const Ip6::Messa
         child->SetState(Neighbor::kStateParentRequest);
         child->SetDataRequestPending(false);
     }
+    else
+    {
+        if ((TimerMilli::GetNow() - child->GetLastHeard()) < kParentRequestRouterTimeout)
```
[the remark this answers is not in the document]
Comment (zhanglongxia) [reply src/core/thread/mle_router.cpp]: updated

File: src/core/openthread-core-default-config.h [hunk 5]
```diff
@@ -159,6 +159,16 @@
 #define OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_POLLS 4
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_TX_NUM_BCAST
+ *
+ * The transmit number of a broadcast frame in MAC layer.
```
Comment (jwhui) [inline src/core/openthread-core-default-config.h]: suggestion:
```
The number of times each IEEE 802.15.4 broadcast frame is transmitted.

The minimum value is 1.  Values larger than 1 may improve broadcast reliability by increasing redundancy, but may also increase congestion.
```

Comment (zhanglongxia) [reply src/core/openthread-core-default-config.h]: Updated. Please review it at your convenience.

**Pull request merged**